app/scripts/upload_to_play_store.py

```
@@ -67,6 +69,68 @@ def get_credentials():
+    was already consumed). It is a heuristic, not a guarantee: we have not
+    confirmed against the API that an abandoned/uncommitted edit never consumes
+    a version code, so callers should treat a True result as "probably already
+    shipped." Scope it to retries only — on a first attempt the same error is a
+    genuine "version not bumped" misconfig and must fail loudly.
+    """
+    if not isinstance(exc, HttpError):
+        return False
+    status = getattr(getattr(exc, 'resp', None), 'status', None)
+    try:
+        content = exc.content.decode('utf-8', 'replace') if isinstance(exc.content, (bytes, bytearray)) else str(exc.content)
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)